resources/innosetup/setup.iss

```diff
@@ -57,15 +58,38 @@ Name: "webdav"; Description: "WebDAV system configuration"; Types: full; ExtraDi
+
+; Remove legacy AutoStart entry; see: https://github.com/cryptomator/cryptomator-win/issues/33
 Root: HKCU; Subkey: "SOFTWARE\Microsoft\Windows\CurrentVersion\Run\"; ValueType: none; ValueName: "Cryptomator"; Flags: dontcreatekey deletevalue
 
+; Remove legacy ProgID introduced in 1.5.12-beta1
+Root: HKCR; Subkey: "EncryptedCryptomatorVaultFile"; Flags: dontcreatekey deletekey
+
+; Remove legacy ProgIDs from before 1.5.12
+Root: HKCR; Subkey: "CryptomatorMasterkeyFile"; Flags: dontcreatekey deletekey
+Root: HKCR; Subkey: "CryptomatorEncryptedData"; Flags: dontcreatekey deletekey
+
+; .cryptomator filetype: Add extension, MIME type, perceived type, description, icon and command
+Root: HKA; Subkey: "Software\Classes\.cryptomator"; ValueType: string; ValueName: ""; ValueData: "Cryptomator.MasterkeyFile.{#ProgIDVersion}"
+Root: HKA; Subkey: "Software\Classes\.cryptomator"; ValueType: string; ValueName: "Content Type"; ValueData: "application/x-vnd.cryptomator.vault-metadata"
+Root: HKA; Subkey: "Software\Classes\.cryptomator"; ValueType: string; ValueName: "PerceivedType"; ValueData: "text"
+
+Root: HKA; Subkey: "Software\Classes\Cryptomator.MasterkeyFile.{#ProgIDVersion}"; ValueType: string; ValueName: ""; ValueData: "Cryptomator Masterkey File"; Flags: deletekey uninsdeletekey
+Root: HKA; Subkey: "Software\Classes\Cryptomator.MasterkeyFile.{#ProgIDVersion}\DefaultIcon"; ValueType: string; ValueName: ""; ValueData: "{app}\Cryptomator.exe,0"
+Root: HKA; Subkey: "Software\Classes\Cryptomator.MasterkeyFile.{#ProgIDVersion}\shell\open\command"; ValueType: string; ValueName: ""; ValueData: """{app}\Cryptomator.exe"" ""%1"""
+
+; .c9r & .c9s filetypes: Add extension, MIME type, perceived type, description and icon for each
+Root: HKA; Subkey: "Software\Classes\.c9r"; ValueType: string; ValueName: ""; ValueData: "Cryptomator.EncryptedData.{#ProgIDVersion}"
+Root: HKA; Subkey: "Software\Classes\.c9r"; ValueType: string; ValueName: "Content Type"; ValueData: "application/x-vnd.cryptomator.encrypted-data"
+Root: HKA; Subkey: "Software\Classes\.c9r"; ValueType: string; ValueName: "PerceivedType"; ValueData: "system"
+
+Root: HKA; Subkey: "Software\Classes\.c9s"; ValueType: string; ValueName: ""; ValueData: "Cryptomator.EncryptedData.{#ProgIDVersion}"
+Root: HKA; Subkey: "Software\Classes\.c9s"; ValueType: string; ValueName: "Content Type"; ValueData: "application/x-vnd.cryptomator.encrypted-data"
+Root: HKA; Subkey: "Software\Classes\.c9s"; ValueType: string; ValueName: "PerceivedType"; ValueData: "system"
+
+Root: HKA; Subkey: "Software\Classes\Cryptomator.EncryptedData.{#ProgIDVersion}"; ValueType: string; ValueName: ""; ValueData: "Cryptomator Encrypted Data"; Flags: deletekey uninsdeletekey
+Root: HKA; Subkey: "Software\Classes\Cryptomator.EncryptedData.{#ProgIDVersion}\DefaultIcon"; ValueType: string; ValueName: ""; ValueData: "{app}\Cryptomator.exe,0"
```

Comment:
When uninstalling Cryptomator, does this key need to be removed?

Reply:
Same in Lines 78, 79. I guess it might get implicitly deleted via its parent key (line 90 and 77 respectively).

Reply:
Still, I agree that it _might_ be more dev friendly to explicitly add this flag to everything that is intended to be deleted during uninstall.

Reply:
Subkeys are in fact implicitly deleted by their parents. 

Therefore this is not the case:
> When uninstalling Cryptomator, does this key need to be removed?

--

> Still, I agree that it might be more dev friendly to explicitly add this flag to everything that is intended to be deleted during uninstall.

IMO the grouping does enough to highlight this fact and adding the flags manually, just adds additional complexity.

Reply:
> IMO the grouping does enough to highlight this fact and adding the flags manually, just adds additional complexity.

Disagree. I don't see any complexity here, the line is already present anyway. Being explicit about something that software is supposed to do, never hurts. Instead it may confuse people (as proven by @infeo's initial comment) who don't know the recursive deletion behaviour of `uninsdeletekey` by heart.

Reply:
Alright, I will add the additional flags in the evening

Reply:
Resolved by 2447ddb